### PR TITLE
Support a non-string name in Series' boxplot

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR fixes `Series.boxplot` to support non-string name in Series's boxplot.

```python
from databricks import koalas as ks
ks.Series([1, 2, 3]).plot.box()
```

**Before:**

```python
<matplotlib.axes._subplots.AxesSubplot object at 0x12251a340>
```

**After:**

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../koalas/databricks/koalas/plot.py", line 1654, in box
    return self(kind="box", **kwds)
  File "/.../koalas/databricks/koalas/plot.py", line 1340, in __call__
    return plot_series(data=self.data, kind=kind, **kwds)
  File "/.../koalas/databricks/koalas/plot.py", line 892, in plot_series
    return _plot(
  File "/.../koalas/databricks/koalas/plot.py", line 1117, in _plot
    plot_obj.generate()
  File "/usr/local/lib/python3.8/site-packages/pandas/plotting/_matplotlib/core.py", line 269, in generate
    self._compute_plot_data()
  File "/.../koalas/databricks/koalas/plot.py", line 317, in _compute_plot_data
    col_stats, col_fences = KoalasBoxPlot._compute_stats(data, colname, whis, precision)
  File "/.../koalas/databricks/koalas/plot.py", line 434, in _compute_stats
    F.mean(colname).alias("{}_mean".format(colname))
  File "/.../spark/python/pyspark/sql/functions.py", line 63, in _
    jc = getattr(sc._jvm.functions, name)(_to_java_column(col))
  File "/.../spark/python/pyspark/sql/column.py", line 45, in _to_java_column
    raise TypeError(
TypeError: Invalid argument, not a string or column: None of type <class 'NoneType'>. For column literals, use 'lit', 'array', 'struct' or 'create_map' function.
```

### CODE REVIEW

File: databricks/koalas/plot.py [hunk 1]
```diff
@@ -300,6 +300,7 @@ def _plot(self, ax, bxpstats, column_num=None, return_type="axes", **kwds):
 
     def _compute_plot_data(self):
         colname = self.data.name
```
Comment (itholic) [inline databricks/koalas/plot.py:302]: Maybe I think we can remove `colname` by changing the variable name below?

```diff
     def _compute_plot_data(self):
-        colname = self.data.name
         spark_column_name = self.data._internal.spark_column_name_for(self.data._column_label)
         data = self.data

@@ -309,7 +308,7 @@ class KoalasBoxPlot(BoxPlot):
         # Gets some important kwds
         showfliers = self.kwds.get("showfliers", False)
         whis = self.kwds.get("whis", 1.5)
-        labels = self.kwds.get("labels", [colname])
+        labels = self.kwds.get("labels", [spark_column_name])
```

**Review by itholic — APPROVED** [review]
Otherwise, LGTM

File: databricks/koalas/plot.py (hunk 1 above)
Comment (HyukjinKwon) [reply databricks/koalas/plot.py:302]: Nope, `labels` should be the user-facing label. This is shown in the output plot image for example.

Comment (itholic) [reply databricks/koalas/plot.py:302]: @HyukjinKwon 
Oh, Sorry I should have checked the visual output on the notebook not only unittest.
Thanks for the correction !

File: databricks/koalas/plot.py [hunk 2]
```diff
@@ -427,7 +430,7 @@ def _compute_stats(data, colname, whis, precision):
         pdf = data._kdf._internal.resolved_copy.spark_frame.agg(
             *[
                 F.expr(
-                    "approx_percentile({}, {}, {})".format(colname, q, int(1.0 / precision))
+                    "approx_percentile(`{}`, {}, {})".format(colname, q, int(1.0 / precision))
```
Comment (ueshin) [inline databricks/koalas/plot.py:433]: Good catch.

Also we might need to surround the `colname` in `_outliers`, `_calc_whiskers`, and `_get_fliers`.

- `F.mean(colname).alias("{}_mean".format(colname))`
- `expression = F.col(colname).between(lfence, ufence)`
- `outliers.filter("not __{}_outlier".format(colname))`
- `.agg(F.min(colname).alias("min"), F.max(colname).alias("max"))`
- `fliers_df = outliers.filter("__{}_outlier".format(colname))`

**Review by ueshin — APPROVED** [review]
LGTM, pending tests.

**Pull request merged**